crates/bevy_reflect/bevy_reflect_derive/src/lib.rs

```diff
@@ -1,680 +1,71 @@
 extern crate proc_macro;
```

Comment:
Maybe consider a doc comment for the crate? Not sure if it's necessary but it could be helpful to have a basic overview of what the most important parts are. E.g. "This crate drives the `#[derive(Reflect)]` macro, these are the macro definitions, other important files are ...", etc.

Reply:
Yep, module level docs are great. Not essential to this PR, but I won't object.

Reply:
Yeah I can try to add a simple summary of the crate and a list of important structs, functions, etc.

It can then be further refined in a separate PR.